### ISSUE TITLE
How to add Cordova-Plugins

### ISSUE THREAD

**Issue by khayalan-mathew** [unimonkiez/react-cordova-boilerplate#15]
Hey guys,

how i can add Cordova-Plugins and use them :?
Because even if i installed them the console says, that for example FCMPlugin isnt defined.
How can i fix that?

**Comment by unimonkiez** [comment]
Which console?  
debugging the server application (which your started with `npm start` command) or debugging a cordova application?

**Comment by khayalan-mathew** [comment]
So i builded it with cordova and emulated it on a android-device, when clicking the button nothing happens ^^ [Indication of error]
When i build it for browser and im activating the devtools i can see, that its not defined.

When i remove the line everything work as intended (except the thing i want to do with the Plugin)

**Comment by unimonkiez** [comment]
Any plugin you install will not work on the browser, it's not built for that.

You should debug on your mobile device, maybe the plugin didn't install correctly.

Usually I wrap any functionality in a seperate file and mock it incase I'm developing on desktop (which is easier to do).
For example, I downloaded a plugin for `Bluetooth communication` and it adds a global named `bluetoothPairToId` so instead of using it directly I wrap it like so:
```javascript
// bluetooth.pair.js
export default (deviceId, cb) => {
  if (__CORDOVA__) {
    window.bluetoothPairToId(deviceId, (err, id) => {
      // ...
      cb({ id: id });
    });
  } else {
    cb({ id: 'MOCK' });
  }
};
```

**Comment by renanbandeira** [comment]
Btw, at the project I work, we created a module to deal with plugins. It is
easier to manage and we can simply dispatch an action to call some plugin
function

**Comment by khayalan-mathew** [comment]
Thanks for the help,

but i didnt get it work... The one thing i search is how i can reference to the PlugIn
Currently im using following Script
`window.FCMPlugin.getToken(function(token) {
      console.debug(token);
    });`
But i dont get whats wrong with it and i cant see any logs in the emulator or :?

**Comment by renanbandeira** [comment]
You should activate dev tools when you run it on the android device and you can see the error.

**Comment by khayalan-mathew** [comment]
The Redux-DevTools ? I cant resize these in the emulator

**Comment by renanbandeira** [comment]
1- Connect your device into your computer
2- Go to chrome://inspect/devices
3- Click to inspect the webview of your device. Check the log if the plugin is getting called.. Btw, I think FCM Plugin should delaay a little bit and I'm not sure if console.debug is going to work even if you are running a deployed apk.

The other way to check if something is happenning is to open Android Studio and check the Android monitor if something related to FCM is getting logged.

**Comment by khayalan-mathew** [comment]
So ^^ The Console says, that it cannot read property "getToken" of undefined... So FCMPlugin is still undefined. I will test it with other plugins

**Comment by renanbandeira** [comment]
So probably your plugin isn't installed correctly

**Comment by khayalan-mathew** [comment]
Tried it with another plugin (cordova-plugin-dialogs) and the same error again ^^
I will post a link to my project down below, if this can help solving the error [Checkout the featúre/react-redux branch] for the actual version (master branch is really old)[ >PROJECT<](https://github.com/khayalan-mathew/gymdonapp/tree/feature/react-redux)

Im going to update it in a few mins ^^ Didnt pushed it in the last few days :)

**Comment by renanbandeira** [comment]
So, I imagine the platforms and plugins folders are ignored by .gitignore. Nevertheless, I can see that your config.xml is almost like the default. Try running the following commands before building:
- `cordova plugin add cordova-plugin-fcm cordova-plugin-dialogs --save`
- `cordova prepare`
Check on plugins folder if cordova-plugin-fcm and cordova-plugin-dialogs folders exist.
Run your app with `cordova run android`

**Comment by khayalan-mathew** [comment]
Updated my repository and the folders still exist ^^ You can find the line, where i call the plugin in the Login.jsx file (https://github.com/khayalan-mathew/gymdonapp/blob/feature/react-redux/src/containers/Login.jsx#L47).
Folders for plugins are there ^^ I will try it in a second with your options :)

**Comment by khayalan-mathew** [comment]
Still the same error *meeh*

**Comment by renanbandeira** [comment]
How is your config.xml after the commands I gave you?

**Comment by khayalan-mathew** [comment]
`<?xml version='1.0' encoding='utf-8'?>
<widget id="one.mathew.gymdon" version="1.1.4" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <hook src="hooks/build.sh" type="before_prepare" />
    <name>GymDon</name>
    <description>
        App for the Gymnasium Donauwörth
    </description>
    <author email="dev@cordova.apache.org" href="http://cordova.io">
        Apache Cordova Team
    </author>
    <content src="index.html" />
    <plugin name="cordova-plugin-whitelist" version="1" />
    <access origin="*" />
    <allow-intent href="http://*/*" />
    <allow-intent href="https://*/*" />
    <allow-intent href="tel:*" />
    <allow-intent href="sms:*" />
    <allow-intent href="mailto:*" />
    <allow-intent href="geo:*" />
    <plugin name="cordova-plugin-fcm" spec="~2.1.1" />
    <plugin name="cordova-plugin-dialogs" spec="~1.3.1" />
</widget>`

**Comment by renanbandeira** [comment]
yeah, you have the plugins insntalled, indeed. Have you tried to use global instead of window in your calls? I don't know what can be, it's just an idea of a brainstorm

**Comment by khayalan-mathew** [comment]
Doesnt change anything :) But thanks for your help

**Comment by khayalan-mathew** [comment]
Found the error :)
Fixed it in #16
